tests/slider.spec.ts

```diff
@@ -0,0 +1,168 @@
+});
+
+test('Displays start and and end values', async ({ page }) => {
+  const sliderMinMax = page.getByTestId('slider-min-max')
+  await expect(sliderMinMax).toBeVisible()
+  const box = await sliderMinMax.boundingBox()
+  await expect(sliderMinMax.getByText(String(customStart), { exact: true })).toBeVisible()
+  const startPercentage = (customStart - customMin) / (customMax - customMin)
+  await expect(sliderMinMax.getByRole('slider').filter({ hasText: String(customStart) })).toHaveCSS('left', String(startPercentage * box.width) + 'px')
+
+  await expect(sliderMinMax.getByText(String(customEnd), { exact: true })).toBeVisible()
+  const endPercentage = (customEnd - customMin) / (customMax - customMin)
+  await expect(sliderMinMax.getByRole('slider').filter({ hasText: String(customEnd) })).toHaveCSS('left', String(endPercentage * box.width) + 'px')
+  
+  // check that slider labels are not visible until on hover
+  const sliderLabels = await sliderMinMax.locator('.floating')
+  await expect((await sliderLabels.all()).length).toBe(2)
+
+  let startLabelOpacity = await sliderMinMax.locator('.floating').first().evaluate((e) => { return getComputedStyle(e).opacity })
+  expect(startLabelOpacity).toBe("0")
+  await sliderMinMax.getByRole('slider').first().hover()
+  await delay(250); // wait for opacity transition
+  startLabelOpacity = await sliderMinMax.locator('.floating').first().evaluate((e) => { return getComputedStyle(e).opacity })
+  expect(startLabelOpacity).toBe("1")
+
+  let endLabelOpacity = await sliderMinMax.locator('.floating').nth(1).evaluate((e) => { return getComputedStyle(e).opacity })
+  expect(endLabelOpacity).toBe("0")
+  await sliderMinMax.getByRole('slider').nth(1).hover()
+  await delay(250); // wait for opacity transition
+  endLabelOpacity = await sliderMinMax.locator('.floating').nth(1).evaluate((e) => { return getComputedStyle(e).opacity })
+  expect(endLabelOpacity).toBe("1")
+});
+
+test('Starts slider at minimum value', async ({ page }) => {
+  const sliderMin = page.getByTestId('slider-min')
+  await expect(sliderMin).toBeVisible()
+  await expect(sliderMin.locator('span.bg-gray-9')).toHaveCSS('left', '0px') // bg-gray-9 is color of slider line
+});
+
+test('Ends slider at maximum value', async ({ page }) => {
+  const sliderMax = page.getByTestId('slider-max')
+  await expect(sliderMax).toBeVisible()
+  const box = await sliderMax.boundingBox()
+  await expect(sliderMax.locator('span.bg-gray-9')).toHaveCSS('right', '0px')
+});
+
+test('Starts slider placement at value', async ({ page }) => {
+  const sliderValue = page.getByTestId('slider-value')
+  await expect(sliderValue).toBeVisible()
+  const box = await sliderValue.boundingBox()
+  const valuePercentage = (value - customMin) / (customMax - customMin)
+  await expect(sliderValue.getByRole('slider')).toHaveCSS('left', String(box.width * valuePercentage) + 'px')
+});
+
+test('Displays axis ticks at intervals of size step', async ({ page }) => {
```

Comment:
It would be good to test this with different step sizes and min maxes. Some really big, others really small, others kinda wacky.

Reply:
We ended up having a weird bug a bit ago with step size 3, min of 0 max of 360. Not sure if it is still there. But might be worth testing